src/classes/ADDR_Seasonal_TEST.cls

```diff
@@ -0,0 +1,923 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2016
+* @group Addresses
+* @group-content ../../ApexDocContent/Addresses.htm
+* @description Tests specific to Seasonal Address Management. 
+*/
+
+@isTest
+private with sharing class ADDR_Seasonal_TEST {
+    
+    /*********************************************************************************************************
+    @description
+        insert new seasonal addresses to Account w/ existing default addresses  
+    verify:
+        contact's && hh address matches new seasonal
+        old default addresses still marked default
+    **********************************************************************************************************/            
+    @isTest
+    public static void insertNewSeasonalAddr() {        
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // create additional addresses
+        addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < 2; i++) {
+            addrs[i].Parent_Account__c = accsAddrs.accs[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].Seasonal_Start_Month__c = String.valueOf(system.today().addMonths(-1).month());
+            addrs[i].Seasonal_Start_Day__c = '1';
+            addrs[i].Seasonal_End_Month__c = String.valueOf(system.today().addMonths(1).month());
+            addrs[i].Seasonal_End_Day__c = '28';
+            addrs[i].MailingStreet__c = 'New Seasonal Street' + i;
+            addrs[i].MailingCity__c = 'New Seasonal City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.stopTest();
+        
+        // verify Account address is new!
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c
+        from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assert(acc.BillingStreet.contains('New Seasonal Street'));
+            system.assert(acc.BillingCity.contains('New Seasonal City'));
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+        
+        // verify the previous addresses still are Default
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c, Latest_Start_Date__c, Latest_End_Date__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.